gtda/local_homology/simplicial.py

```diff
@@ -0,0 +1,411 @@
+from numbers import Real
+from types import FunctionType
+
+import numpy as np
+import warnings
+
+from scipy.spatial.distance import pdist, squareform
+from sklearn.neighbors import KNeighborsTransformer, RadiusNeighborsTransformer
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils import check_array
+from sklearn.utils.validation import check_is_fitted
+
+from gtda.homology import VietorisRipsPersistence
+from gtda.utils.intervals import Interval
+from gtda.utils.validation import validate_params
+from gtda.plotting import plot_diagram
+from gtda.base import PlotterMixin
+from gtda.utils._docs import adapt_fit_transform_docs
+
+
+class LocalVietorisRipsBase(BaseEstimator,
+                            TransformerMixin,
+                            PlotterMixin):
+    """
+    Base class for KNeighboursLocalVietorisRips and RadiusLocalVietorisRips.
+    LocalVietorisRipsBase is not meant to be used. Please see documentation
+    for KNeighboursLocalVietorisRips and RadiusLocalVietorisRips."""
+
+    def __init__(self, metric='euclidean', homology_dimensions=(1, 2),
+                 neighborhoud_param=(1, 2), n_jobs=1):
+        """
+        Initializes the base class by setting the parameters, initializes the
+        object used for computing persistence homology, and checking that the
+        parameters were initialized correctly."""
+
+        # metric for the point cloud
+        self.metric = metric
+
+        # topological dimension of features to be computed
+        self.homology_dimensions = homology_dimensions
+
+        self.n_jobs = n_jobs
+
+        # Object is used to compute persistence diagrams
```

Comment:
Could we move all the non-trivial part of `__init__` (i.e. after the `n_jobs` line) to the `fit` method?  In `scikit-learn`, it is considered bad practice to have anything but trivial assignments to class attributes in the `__init__` method.

Reply:
Done!